src/Functions/h3toGeo.cpp

```diff
@@ -52,9 +65,18 @@ class FunctionH3ToGeo : public IFunction
                 "Illegal type {} of argument {} of function {}. Must be UInt64",
                 arg->getName(), 1, getName());
 
-        return std::make_shared<DataTypeTuple>(
-            DataTypes{std::make_shared<DataTypeFloat64>(), std::make_shared<DataTypeFloat64>()},
-            Strings{"longitude", "latitude"});
+        if (h3togeo_lon_lat_result_order)
+        {
+            return std::make_shared<DataTypeTuple>(
```

Comment:
I think we need to switch both branches, no?

Reply:
I think it's fine now? `h3togeo_lon_lat_result_order` means `longitude` comes first. That's the behaviour now.